src/S3.cpp

```diff
@@ -5,120 +5,12 @@
 using namespace Aws::S3;
 
 Aws::SDKOptions options;
-
+namespace cirrus {
 void s3_initialize_aws() {
```

Comment:
I think we should check whether this functions has been called twice (that shouldn't happen).

Something like:

```
void s3_initialize_aws() {
   static bool called = false;
   if (called == false) {
      called = true;
   } else {
      throw "some error"
   }
   ....
}
```